torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -231,6 +231,13 @@ class FullyShardedDataParallel(nn.Module, _FSDPState):
         thread in that way prevents over-allocating memory for subsequent
         all-gathers, and it should not actually delay GPU kernel execution.
 
+    .. note::
+        When using ``sharding_strategy=ShardingStrategy.HYBRID_SHARD`` with the
+        sharding process group being intra-node and the replication process
+        group being inter-node, setting ``NCCL_CROSS_NIC=1`` can help improve
+        the all-reduce times over the replication process group for some
+        cluster setups.
```

Comment:
@wz337 Since we will be forcing users to use `DeviceMesh` for HSDP (HSDP state_dict only supports `DeviceMesh`), we may want to revisit the docstring later to ensure the correct wording.